stix_shifter_modules/sentinelone/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,411 @@
+        """
+        Formats value in the event of set operation
+        :param values
+        :return formatted value
+        """
+        if filter_name in ["fileFullName"]:
+            raise NotImplementedError(f'IN operator is not supported for File Name, Use LIKE or MATCHES operator only')
+        if filter_name in self.type_map["boolean_supported_fields"]:
+            raise NotImplementedError("IN operator is not supported for Boolean fields")
+        if filter_name in self.type_map["timestamp_supported_fields"]:
+            raise NotImplementedError("IN operator is not supported for Timestamp fields")
+        gen = values.element_iterator()
+
+        if filter_name in self.type_map["enum_supported_fields"]:
+            val_gen = values.element_iterator()
+            for final_value in val_gen:
+                if final_value.upper() not in [x.upper() for x in self.type_map["enum_supported_values"][filter_name]]:
+                    raise NotImplementedError(f'Unsupported ENUM values provided. Possible supported enum values are'
+                                              f'{self.type_map["enum_supported_values"][filter_name]}')
+            formatted_value = ','.join([QueryStringPatternTranslator._escape_value(value.upper())
+                                        for value in gen])
+        else:
+            formatted_value = ','.join([QueryStringPatternTranslator._escape_value(value)
+                                    for value in gen])
+        return f'({formatted_value})'
+
+    def _format_match(self, value, filter_name) -> str:
+        """
+        Formats value in the event of match operation
+        :param value
+        :return formatted string type value
+        """
+        if filter_name not in self.type_map["string_supported_fields"]:
+            raise NotImplementedError("only String fields are supported in MATCH operator")
+        raw = QueryStringPatternTranslator._escape_value(value)
+        return f'{raw}'
+
+    def _format_equality(self, value, filter_name) -> str:
+        """
+        Formats value in the event of equality operation
+        :param value
+        :return formatted value
+        """
+        value = self._check_field_value_support(filter_name, value)
+        value_escaped = QueryStringPatternTranslator._escape_value(value)
+        return f'{value_escaped}'
+
+    def _format_like(self, value, filter_name) -> str:
+        """
+        Formats value in the LIKE of equality operation
+        :param value
+        :return formatted value
+        """
+        if filter_name not in self.type_map["string_supported_fields"]:
+            raise NotImplementedError("only String fields are supported in LIKE operator")
+        wildcard = ['%', '$', '+', '*', '^', '?']
+        for val in wildcard:
+            if val in value:
+                raise NotImplementedError("Wildcard characters is not supported in sentinelone LIKE operator")
```

Comment:
It might be better to log this message rather than raise an error, and then strip out the wildcards. If a LIKE search is being made with multiple connectors at the same time, and one connector does support wildcards, then this connector will raise an error on what is a valid STIX pattern.

Reply:
raising error is removed, now error message is logged and continuing after removing the wildcard character.